api/src/index.js

```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
 app.listen(process.env.PORT, () => {
   console.log(`API listening on port ${process.env.PORT}`);
 });
+
+app.get("/my-route", (req, res) => {
+  res.send("Hi friend");
+});
+
+apiRouter.get("/future-meals", async (req, res) => {
+  try{
+    const GET_FUTURE_MEALS_QUERY = "SELECT * FROM meal WHERE `when` > NOW();";
+    const [futureMeals] = await knex.raw(GET_FUTURE_MEALS_QUERY);
+    res.json(futureMeals);
+  } catch (error) {
+    console.error("Fetching Future Meals Error:", error);
+    res.status(500).json({ error: "Not successful" });
+  }
+});
+
+apiRouter.get("/past-meals", async (req, res) => {
+  try{
+    const GET_PAST_MEALS_QUERY = "SELECT * FROM meal WHERE `when` < NOW();";
+    const [pastMeals] = await knex.raw(GET_PAST_MEALS_QUERY);
+    res.json(pastMeals);
+  } catch (error) {
+    console.error("Fetching Past Meals Error:", error);
+    res.status(500).json({ error: "Not successful" });
+  }
+});
+
+apiRouter.get("/all-meals", async (req, res) => {
+  try{
+    const GET_ALL_MEALS_QUERY = "SELECT * FROM meal ORDER BY ID;";
+    const [allMeals] = await knex.raw(GET_ALL_MEALS_QUERY);
+    res.json(allMeals);
+  } catch (error) {
+    console.error("Fetching All Meals Error:", error);
+    res.status(500).json({ error: "Not successful" });
+  }
+});
+
+apiRouter.get("/first-meal", async (req, res) => {
+  try{
+    const GET_FIRST_MEAL_QUERY = "SELECT * FROM meal ORDER BY ID LIMIT 1;";
+    const [firstMeal] = await knex.raw(GET_FIRST_MEAL_QUERY);
+
+    if(!firstMeal || firstMeal.length === 0) {
+      return res.status(404).json({ error: "No available meals."});
+    }
+    res.json(firstMeal);
```

Comment:
```suggestion
    res.json(firstMeal[0]);
```
Even though there is only one row (meal) returned, the type of the variable is still an array, meaning you have to dereference the first value (`[0]`) to make sure you return an object, not an array.